pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -63,8 +63,8 @@ type LaunchTemplateProvider struct {
 	clusterEndpoint       string
 }
 
-func NewLaunchTemplateProvider(ctx context.Context, ec2api ec2iface.EC2API, amiFamily *amifamily.Resolver, securityGroupProvider *securitygroup.Provider, caBundle *string, startAsync <-chan struct{}, kubeDNSIP net.IP, clusterEndpoint string) *LaunchTemplateProvider {
-	l := &LaunchTemplateProvider{
+func NewProvider(ctx context.Context, ec2api ec2iface.EC2API, amiFamily *amifamily.Resolver, securityGroupProvider *securitygroup.Provider, caBundle *string, startAsync <-chan struct{}, kubeDNSIP net.IP, clusterEndpoint string) *Provider {
+	l := &Provider{
```

Comment:
How about passing the cache through dependency injection rather than having to expose it through public API?